.travis.yml

```diff
@@ -14,19 +14,25 @@ addons:
     - gfortran
     - libncurses5-dev
 
-before_install:
-  - pip install pytest-cov
-  - pip install pytest-ordering
-  - pip install coveralls
-  - pip install future
-  - pip install pysatCDF >/dev/null
-  # Travis CI currently not working with netCDF4 1.5.3
-  - pip install 'cftime==1.1.1'
-  - pip install 'netCDF4<1.5.3'
-
-# command to install dependencies
 install:
-  - source activate test-environment
+  - sudo apt-get update
+  - wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh
+  - bash miniconda.sh -b -p $HOME/miniconda
+  - source "$HOME/miniconda/etc/profile.d/conda.sh"
+  - hash -r
+  - conda config --set always_yes True --set changeps1 False
+  - conda update -q conda
+  # Useful for debugging any issues with conda
+  - conda info -a
+  - conda create -q -n test-environment python=$TRAVIS_PYTHON_VERSION numpy scipy pandas xarray requests beautifulsoup4 lxml netCDF4 h5py nose
+  - conda activate test-environment
+  - conda install pytest-cov
+  - conda install pytest-ordering
+  - conda install coveralls
+  - conda install future
+  - pip install madrigalWeb
+  - pip install PyForecastTools
+  - pip install pysatCDF >/dev/null
```

Comment:
Comment explaining why conda is used for some and pip for others would be useful.